checks/check-slither.ts

```diff
@@ -0,0 +1,78 @@
+import { getContractName } from '../utils/clients/tenderly'
+import { ProposalCheck } from '../types'
+
+// convert exec method from a callback to a promise
+const exec = util.promisify(execCallback)
+
+// Data returned from command execution
+type ExecOutput = {
+  stdout: string
+  stderr: string
+}
+
+/**
+ * Runs slither against the verified contracts and reports the outputs.
+ * Assumes slither is already installed and can be run with python
+ */
+export const checkSlither: ProposalCheck = {
+  name: 'Runs slither against the verified contracts',
+  async checkProposal(proposal, sim, deps) {
+    let info = ''
+    let warnings: string[] = []
+    // For each verified contract with code, we write the files, run slither, then delete the
+    // files. We do this instead of running Slither on all contracts at once since contracts at
+    // different addresses can have the same names, which would result in an error when compiling
+    for (const contract of sim.contracts) {
+      // Get solc version used
+      const solcVersionMatch = contract.compiler_version.match(/\d*\.\d*\.\d*/)
+      if (!solcVersionMatch) {
+        const msg = `Slither not run for \`${contract.contract_name}\` at \`${contract.address}\`: could not parse solc version`
+        warnings.push(msg)
+        continue
+      }
+      const solcVersion = solcVersionMatch[0]
+
+      // Save the contract locally
+      for (const file of contract.data.contract_info) writeFileSync(file.name, file.source)
+
+      // Run slither against it
+      const output = await runSlither(solcVersion)
+      if (!output) {
+        warnings.push(`Slither execution failed for \`${contract.contract_name}\` at \`${contract.address}\``)
+        continue
+      }
+
+      // Append results to report info
+      // Note that slither supports a `--json` flag  we could use, but directly printing the formatted
+      // results in a code block is simpler and sufficient for now
+      info += `\n - Slither report for ${getContractName(contract)}`
+      info += `\n\`\`\`\n${output.stderr}\`\`\``
+
+      // Delete the contract files
+      for (const file of contract.data.contract_info) unlinkSync(file.name)
+    }
+
+    return { info: [info], warnings, errors: [] }
+  },
+}
+
+/**
+ * Tries to run slither via python installation
+ * @dev Requires solc-select and slither to be installed
+ * @dev If you have nix/dapptools installed, you'll need to make sure the path to your python
+ * executables (find this with `which solc-select`) comes before the path to your nix executables.
+ * This may require editing your $PATH variable prior to running this check. If you don't do this,
+ * the nix version of solc will take precedence over the solc-select version, and slither will fail
+ */
+async function runSlither(solcVersion: string): Promise<ExecOutput | null> {
+  try {
+    return await exec(`solc-select install ${solcVersion} && SOLC_VERSION=${solcVersion} slither .`)
```

Comment:
you don't have to return here

Reply:
maybe this is now outdated but pretty sure you need both `return` and `await` here: https://stackoverflow.com/questions/38708550/difference-between-return-await-promise-and-return-promise

Reply:
oops i meant you don't have to await lol

Reply:
heh yea thought so, still seems both are needed here though

Reply:
yeah fair